central/group/service/service_impl.go

```diff
@@ -87,7 +93,7 @@ func (s *serviceImpl) GetGroups(ctx context.Context, req *v1.GetGroupsRequest) (
 }
 
 func (s *serviceImpl) GetGroup(ctx context.Context, props *storage.GroupProperties) (*storage.Group, error) {
-	group, err := s.groups.Get(ctx, props)
+	group, err := s.groups.Get(ctx, props.GetId())
```

Comment:
This might break current clients who don't know about ID yet (same for `DeleteGroup()`). To avoid breaking the API, let's:
* iff `id` is absent, search the store using `props` and log a warning (maybe one day we'll be adding [deprecation headers](https://tools.ietf.org/id/draft-dalal-deprecation-header-00.html#rfc.section.2))
* iff there is at most one object found, return it
* else, i.e., more than one group is associated with `props`, return `400`
* leave a note to remove that code after the deprecation cycle.

Reply:
I've addressed this (to be precise, not only for `Get, Delete` but also for `Update, Mutate`). 
This is handled in the store layer instead of the datastore layer, providing the same behavior that you have previously mentioned.